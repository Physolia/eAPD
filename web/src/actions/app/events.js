import axios from '../../util/api';
import { selectApdData } from '../../reducers/apd.selectors';

export const saveApdEvent =
  (eventType, metadata = null) =>
  (dispatch, getState) => {
    const state = getState();
    const { id: apdID } = selectApdData(state);

    return axios
      .post(`/apds/${apdID}/events`, { eventType, metadata })
      .then(res => (res.data ? res.data : null));
  };

// Stryker disable next-line ObjectLiteral
export default { saveApdEvent };
