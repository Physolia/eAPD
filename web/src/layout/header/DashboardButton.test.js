import { shallow } from 'enzyme';
import React from 'react';
import {
  plain as DashboardButton,
  mapDispatchToProps
} from './DashboardButton';
import { goToDashboard } from '../../actions/app';

describe('the dashboard button', () => {
  it('renders as expected', () => {
    const dashboard = jest.fn();

    expect(
      shallow(
        <DashboardButton dashboard={dashboard}>clicky clicky</DashboardButton>
      )
    ).toMatchSnapshot();
  });

  it('dispatches the dashboard action when clicked', () => {
    const dashboard = jest.fn();
    const component = shallow(
      <DashboardButton dashboard={dashboard}>clicky</DashboardButton>
    );

    component.simulate('click');

    expect(dashboard).toHaveBeenCalled();
  });

  it('maps redux actions to props', () => {
    expect(mapDispatchToProps).toEqual({ dashboard: goToDashboard });
  });
});
