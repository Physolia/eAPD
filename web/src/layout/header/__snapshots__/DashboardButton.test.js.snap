// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the dashboard button renders as expected 1`] = `
<Button
  component="button"
  onClick={[Function]}
  type="button"
  variation="transparent"
>
  clicky clicky
</Button>
`;
